pkg/config/json_test.go

```diff
@@ -62,55 +55,38 @@ var jsonTests = map[string]jsonTest{
 		newVM: func(t *testing.T) *VirtualMachine {
 			vm := newLinuxVM(t)
 			// virtio-serial
-			dev, err := VirtioSerialNew("/virtioserial")
-			require.NoError(t, err)
-			err = vm.AddDevice(dev)
-			require.NoError(t, err)
+			dev := VirtioSerialNew("/virtioserial")
+			vm.AddDevice(dev)
 			// virtio-input
-			dev, err = VirtioInputNew(VirtioInputKeyboardDevice)
-			require.NoError(t, err)
-			err = vm.AddDevice(dev)
+			dev, err := VirtioInputNew(VirtioInputKeyboardDevice)
 			require.NoError(t, err)
+			vm.AddDevice(dev)
 			// virtio-gpu
-			dev, err = VirtioGPUNew()
-			require.NoError(t, err)
-			err = vm.AddDevice(dev)
+			dev = VirtioGPUNew()
 			require.NoError(t, err)
+			vm.AddDevice(dev)
 			// virtio-net
 			dev, err = VirtioNetNew("00:11:22:33:44:55")
```

Comment:
This is the main reason why most `*New()` methods return an error, some of them require an error return, so for me it's better if they are consistent and all return an error, instead of having to remember it. This also makes the API future-proof if one day we need to return an error.